File: test-automation-core/src/main/java/com/nortal/test/core/services/report/html/Commit.java
package com.nortal.test.core.services.report.html;

import java.time.Instant;
import java.util.List;
import java.util.stream.Collectors;
import org.eclipse.jgit.diff.DiffEntry;
import org.eclipse.jgit.revwalk.RevCommit;
import lombok.Data;
import lombok.Value;
import static java.time.temporal.ChronoUnit.DAYS;

@Data
public class Commit {
	private static final String ROOT_PATH = "dcp-test-automation-data/src/main/resources/";
	private final RevCommit commit;
	private final List<DiffEntry> diffsEntries;

	public String getAuthor() {
		return commit.getAuthorIdent().getName();
	}

	public String getAgo() {
		return DAYS.between(Instant.ofEpochSecond(commit.getCommitTime()), Instant.now()) + " days ago";
	}

	public String getSha() {
		return commit.getName();
	}

	public String getMessage() {
		return commit.getShortMessage();
	}

	public List<Diff> getDiffs() {
		return diffsEntries.stream()
				.map(Diff::new)
				.collect(Collectors.toList());
	}

	@Value
	public static class Diff {
		String path;
		String op;

		public Diff(final DiffEntry diff) {
			this.path = resolvePath(diff).replaceAll(ROOT_PATH, "");
			this.op = diff.getChangeType().toString();
		}

		private String resolvePath(final DiffEntry diff) {
			switch (diff.getChangeType()) {
			case ADD:
				return diff.getNewPath();
			case COPY:
			case RENAME:
				return diff.getOldPath() + "->" + diff.getNewPath();
			case DELETE:
			case MODIFY:
				return diff.getOldPath();
			default:
				throw new IllegalStateException("Unexpected value: " + diff.getChangeType());
			}
		}

	}

}

File: test-automation-core-old/src/main/java/com/nortal/test/core/services/report/html/Commit.kt
package com.nortal.test.core.services.report.html

import lombok.Data
import lombok.Value
import org.eclipse.jgit.diff.DiffEntry
import org.eclipse.jgit.revwalk.RevCommit
import java.time.temporal.ChronoUnit
import java.lang.IllegalStateException
import java.time.Instant
import java.util.stream.Collectors

@Data
class Commit {
    private val commit: RevCommit? = null
    private val diffsEntries: List<DiffEntry>? = null
    val author: String
        get() = commit!!.authorIdent.name
    val ago: String
        get() = ChronoUnit.DAYS.between(Instant.ofEpochSecond(commit!!.commitTime.toLong()), Instant.now()).toString() + " days ago"
    val sha: String
        get() = commit!!.name
    val message: String
        get() = commit!!.shortMessage
    val diffs: List<Diff>
        get() = diffsEntries!!.stream()
            .map { diff: DiffEntry -> Diff(diff) }
            .collect(Collectors.toList())

    @Value
    class Diff(diff: DiffEntry) {
        var path: String
        var op: String

        init {
            path = resolvePath(diff).replace(ROOT_PATH.toRegex(), "")
            op = diff.changeType.toString()
        }

        private fun resolvePath(diff: DiffEntry): String {
            return when (diff.changeType) {
                DiffEntry.ChangeType.ADD -> diff.newPath
                DiffEntry.ChangeType.COPY, DiffEntry.ChangeType.RENAME -> diff.oldPath + "->" + diff.newPath
                DiffEntry.ChangeType.DELETE, DiffEntry.ChangeType.MODIFY -> diff.oldPath
                else -> throw IllegalStateException("Unexpected value: " + diff.changeType)
            }
        }
    }

    companion object {
        private const val ROOT_PATH = "dcp-test-automation-data/src/main/resources/"
    }
}